src/Common/src/TypeSystem/Common/TypeHashingAlgorithms.cs

```diff
@@ -138,11 +138,28 @@ public static int ComputeGenericInstanceHashCode<ARG>(int genericDefinitionHashC
         /// <param name="typeHashcode">Hashcode of the type that owns the method</param>
         /// <param name="nameOrNameAndGenericArgumentsHashcode">Hashcode of either the name of the method (for non-generic methods) or the GenericInstanceHashCode of the name+generic arguments of the method.</param>
         /// <returns></returns>
-        public static int ComputeMethodHashcode(int typeHashcode, int nameOrNameAndGenericArgumentsHashcode)
+        public static int ComputeMethodHashCode(int typeHashcode, int nameOrNameAndGenericArgumentsHashcode)
         {
             // TODO! This hash combining function isn't good, but it matches logic used in the past
             // consider changing to a better combining function once all uses use this function
             return typeHashcode ^ nameOrNameAndGenericArgumentsHashcode;
         }
+
+        /// <summary>
+        /// Produce a hashcode for a generic signature variable
+        /// </summary>
+        /// <param name="index">zero based index</param>
+        /// <param name="method">true if the signature variable describes a method</param>
+        public static int ComputeSignatureVariableHashCode(int index, bool method)
+        {
+            if (method)
+            {
+                return index * 0x7822381 + 0x54872645;
```

Comment:
We only have about 2 bits of entropy here in any case. There's nothing really to do other than make sure that merging this value with something else doesn't actively harm the other hash